retrieval/aggregator.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retrieval
```

Comment:
Can you add a basic explanation of how the aggregator works? It should at least say that it aligns the counters to a known interval (what is it?) and outputs a counter metric with delta encoding, i.e. where the intervals are non-overlapping.

Reply:
The aggregator does not actually handle alignment, it's taken care of by the Stackdriver OpenCensus exporter.

I believe, the Stackdriver OpenCensus exporter uses regular cumulative counters rather than pseudo delta metrics: https://github.com/Stackdriver/stackdriver-prometheus-sidecar/blob/f6e3560dd5068402894da542d1f6c20f9f93baaa/vendor/contrib.go.opencensus.io/exporter/stackdriver/stats.go#L250